gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -443,9 +443,10 @@ OGRErr OGRFlatGeobufLayer::querySpatialIndex() {
         VSIFree(m_featureOffsets);
         m_featureOffsets = nullptr;
     }
-    auto featureOffsetsSize = static_cast<size_t>(featuresCount) * 8;
+    auto featureOffetsCount = static_cast<size_t>(featuresCount);
+    auto featureOffsetsSize = featureOffetsCount * 8;
     m_featureOffsets = static_cast<uint64_t *>(VSI_MALLOC_VERBOSE(featureOffsetsSize));
-    if (VSIFReadL(m_featureOffsets, 8, static_cast<size_t>(featuresCount), m_poFp) != featuresCount)
+    if (VSIFReadL(m_featureOffsets, 8, featureOffetsCount, m_poFp) != featureOffetsCount)
```

Comment:
the above malloc() could potential fail, so null checking should probably be done